### PR TITLE
conbench/runner.py: fix 'name' treatment, clarify

### PR DESCRIPTION
**Pull request by jgehrcke**
Tiny follow-up: https://github.com/conbench/conbench/pull/1043#discussion_r1158183990, and some clarifications in the surrounding code context.

### CODE REVIEW

**Comment by jgehrcke** [comment]
Feedback appreciated, :ship:!

File: conbench/runner.py [hunk 1]
```diff
@@ -246,11 +264,14 @@ def record(self, result, name, error=None, publish=True, **kwargs):
             output,
         ) = self._init(kwargs)
 
-        if (tags.get("name") is not None) and (name == tags["name"]):
+        if (tags.get("name") is not None) and (name != tags["name"]):
```
Comment (austin3dickey) [inline conbench/runner.py:267]: Thank you! Agreed this was the intent

Comment (jonkeane) [reply conbench/runner.py:267]: yes, this is the right intent — should we add a test to cover this?

Comment (austin3dickey) [reply conbench/runner.py:267]: I can add a test when I bump the version. Coming soon.

**Pull request merged**